File: UnitySokoban/Assets/Scripts/Planning/HeuristicSearchPlannerSGW/HeuristicNode.cs
using StateSpaceSearchProject;

namespace HeuristicSearchPlannerSGW
{
    public class HeuristicNode
    {
        private static int nextID = 0;

        int id = nextID++;
        public readonly StateSpaceNode state;
	    public readonly double heuristic;

        internal HeuristicNode(StateSpaceNode state, double heuristic)
        {
            this.state = state;
            this.heuristic = heuristic;
        }
    }
}

File: Heuristic Search (sgware)/src/edu/uno/ai/planning/h/HeuristicNode.java
package edu.uno.ai.planning.h;

import edu.uno.ai.planning.ss.StateSpaceNode;

public class HeuristicNode {

	private static int nextID = 0;
	
	final int id = nextID++;
	public final StateSpaceNode state;
	public final double heuristic;
	
	HeuristicNode(StateSpaceNode state, double heuristic) {
		this.state = state;
		this.heuristic = heuristic;
	}
}
